### PR TITLE
fixing bugs

### PR DESCRIPTION
**Pull request by moontrip**
This will address the first two issues reported https://github.com/VEuPathDB/web-eda/issues/1350
In addition, this PR addressed the dependent axis range of Barplot Viz for Proportion mode (0 to 1 or minPos to 1 for log scale), suggested at https://github.com/VEuPathDB/web-eda/issues/1099.

For the same examples shown in the ticket, here are new screenshot after fixing

a) proportion and logscale on:
histogram-bugfix1

b) incorrect truncation with proportion:
histogram-bugfix2

### CODE REVIEW

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 1]
```diff
@@ -445,17 +481,19 @@ function BarplotViz(props: VisualizationProps) {
...
       truncationConfig(
         {
-          ...(minPos != null && max != null
-            ? {
-                dependentAxisRange: { min: minPos, max: max },
-              }
+          ...(dependentMinPosMax != null &&
+          dependentMinPosMax.min != null &&
+          dependentMinPosMax.max != null
+            ? { dependentAxisRange: dependentMinPosMax }
```
Comment (bobular) [inline src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:487]: I'll have another look at this tomorrow, but are you sure `truncationConfig` needs new 5th and 6th arguments?

If we pass `minPosMax` as the first argument and leave 5th and 6th undefined, I think it works fine.  Though I can't be sure. I still haven't understood what causes the bug where setting dependent range max to 0.5 in proportion mode erroneously causes the truncation warning as in #1350 (b) 

I'll look into #1350 (a) more too!

**Review by bobular — COMMENTED** [review]
Ah, yes, now I understand bug (b) properly. You are correct to "separate out `minPosMax` and `dependentMinPosMax`" but I think my suggestion to keep `truncationConfig()` with 4 args and passing `minPosMax` as the 1st is valid.

I'll look into this more tomorrow, including bug (a)!

**Comment by bobular** [comment]
If you could point to where bug (a) is fixed, I'd appreciate it!

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 2]
```diff
@@ -409,16 +409,52 @@ function BarplotViz(props: VisualizationProps) {
     vizConfig.checkedLegendItems
   );
 
-  const minPos = useMemo(() => barplotDefaultDependentAxisMinPos(data), [data]);
-  const max = useMemo(() => barplotDefaultDependentAxisMax(data), [data]);
+  const minPos = useMemo(() => barplotDefaultDependentAxisMinPos(data), [
+    data,
+    vizConfig.valueSpec,
```
[the remark this answers is not in the document]
Comment (moontrip) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:414]: @bobular Hmm I thought I left data dependency only but I might have missed it - also noticed some unnecessary commented out parts so I will clean them up

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 3]
```diff
@@ -409,16 +409,52 @@ function BarplotViz(props: VisualizationProps) {
...
+  ]);
+  const max = useMemo(() => barplotDefaultDependentAxisMax(data), [
+    data,
+    vizConfig.valueSpec,
+  ]);
+  const minPosMax =
+    minPos != null && max != null ? { min: minPos, max: max } : undefined;
+  const dependentMinPosMax = useMemo(() => {
+    return minPosMax != null && minPosMax.max != null && minPosMax.max != null
```
[the remark this answers is not in the document]
Comment (moontrip) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:423]: @bobular another Oops. Will change it 👍

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx (hunk 1 above)
Comment (moontrip) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:487]: @bobular Yes you are correct. Having those 5th and 6th arguments was actually my older approach before using minPosMax separately. I will change them accordingly.

**Comment by moontrip** [comment]
> If you could point to where bug (a) is fixed, I'd appreciate it!

@bobular It is due to no calculation of minPos for histogram Viz: barplot Viz has a logic to compute minPos correctly. Thus, I added a filter to find minPos at https://github.com/VEuPathDB/web-eda/pull/1352/files#diff-0fdb395524fa60acb62e446f7a13459add5714a7f46591a32f536f277fbfc0ec

**Comment by bobular** [comment]
> > If you could point to where bug (a) is fixed, I'd appreciate it!
> 
> @bobular It is due to no calculation of minPos for histogram Viz: barplot Viz has a logic to compute minPos correctly. Thus, I added a filter to find minPos at https://github.com/VEuPathDB/web-eda/pull/1352/files#diff-0fdb395524fa60acb62e446f7a13459add5714a7f46591a32f536f277fbfc0ec

Thank you. I thought I had tried disabling that fix and still saw the correct axis ticks, but I've just done it again carefully and can confirm this fixes the problem (the histogram back end *does* return zero count bins, I thought maybe it didn't).

If you could let me know when to do a final review, I'd appreciate it.

**Comment by moontrip** [comment]
@bobular I have already updated with your suggestions so you can review when you are available 👍

**Review by bobular — APPROVED** [review]
Looks and works great. Required a merge from main to bring in the latest web-components with "not allowed equal" range behaviour.

**Comment by moontrip** [comment]
@bobular Thank you for your reviews/tests and your kind update 👍

**Pull request merged**